### PR TITLE
CR-1109170: sw_emu pl kernel debug in GDB

### PR DESCRIPTION
**Pull request by vboggara-xilinx**
CR-1109170: sw_emu pl kernel debug in GDB

Implemented automated flow for PL Kernel Debug through command line in sw_emu.

### CODE REVIEW

File: src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx [hunk 1]
```diff
@@ -476,9 +476,20 @@ namespace xclcpuemhal2 {
             childArgv[5] = portStream.str().c_str() ;
           }
         }
-        int r = execl(modelDirectory.c_str(), childArgv[0], childArgv[1],
+
+
+        int r = 0;
+
+        if (xclemulation::is_sw_emulation() && xrt_core::config::get_flag_sw_emu_kernel_debug()){// Launch sw_emu device Process in GDB -> Emulation.kernel-dbg =true
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:483]: why is is_sw_emulation check inside cpu_em driver?

Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:483]: @chvamshi-xilinx , its additional check only, nothing harm.

Comment (chvamshi-xilinx) [reply src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:483]: yaa. It might be confusing in future. better to delete :-)

Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:483]: sure @chvamshi-xilinx ,will remove this additional check in next PR.

Comment (chvamshi-xilinx) [reply src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:483]: Thanks @vboggara-xilinx

**Pull request merged**